browser-extension/src/rateLaterHistory.js

```diff
@@ -178,6 +186,42 @@ const addRateLaterBulk = async (videoIds) =>
     );
   });
 
+const skipPreviouslyImported = async (videoSet) => {
+  const newVideos = new Set();
+  const localStorage = await chrome.storage.local.get('youtubeHistoryImported');
+  const historyInStorage = localStorage?.youtubeHistoryImported;
+
+  for (const video of videoSet) {
+    if (!historyInStorage?.includes(video)) {
+      newVideos.add(video);
+    }
+  }
+
+  return newVideos;
+};
+
+const saveImportedToLocalStorage = async (importedVideosSet) => {
+  const localStorage = await chrome.storage.local.get('youtubeHistoryImported');
+  const historyInStorage = localStorage?.youtubeHistoryImported;
+
+  let videosStr = '';
+  for (const video of importedVideosSet) {
+    if (!historyInStorage?.includes(video)) {
+      videosStr += `,${video}`;
+    }
```

Comment:
Loading `youtubeHistoryImported` in a `Set` could perform better, instead of using `String.includes()`. I suppose that is not issue even for thousands of videos?

Reply:
Yes I thought about it. I tried with a string of 10,000 videos and the performance were good enough, but it's probably better to use a `Set`.